src/marin/rl/evaluate_environment.py

```diff
@@ -25,31 +25,63 @@
 import json
 
 from typing import Any
-from dataclasses import dataclass
+from dataclasses import dataclass, asdict
 from levanter.trainer import TrainerConfig
-from levanter.models.llama import LlamaConfig
+from levanter.models.lm_model import LmConfig
 from ray.runtime_env import RuntimeEnv
-from transformers import AutoConfig, AutoTokenizer
+from transformers import AutoTokenizer
 from levanter.inference.openai import InferenceServer, InferenceServerConfig
 from levanter.inference.engine import InferenceEngineConfig
+from levanter.infra.ray_tpu import run_on_pod_ray
 
 from marin.resources import TpuPodConfig
 from marin.training.training import _add_run_env_variables
 from marin.execution import ExecutorStep
 from marin.rl.environments.base import MarinEnv, EnvConfig, load_environment_from_spec
 from marin.rl.model_utils import load_model_from_checkpoint
 from marin.rl.rollout_worker import InferenceContext
+from marin.rl.types import Rollout, RolloutGroup
 from marin.utils import remove_tpu_lockfile_on_exit
 
 logger = logging.getLogger("ray")
 
 
+def _to_list(arr) -> list:
+    """Convert array-like object to list, handling both JAX arrays and Python lists."""
+    if isinstance(arr, list):
+        return arr
+    elif hasattr(arr, "tolist"):
+        return arr.tolist()
+    else:
+        # Fallback for other array types
+        return list(arr)
+
+
+def rollout_to_dict(rollout: "Rollout") -> dict[str, Any]:
```

Comment:
could replace this and rollout group conversion below with:

dataclasses.asdict(
  jax.tree.map(lambda v: _as_list(v) if isinstance(v, list|jax.Array|numpy.ndarray) else v, rollout_group))

That would be a bit more robust if we e.g. add new fields to the module.